src/mumble/GlobalShortcut_win.cpp

```diff
@@ -229,18 +320,22 @@ void GlobalShortcutWin::injectKeyboardMessage(MSG *msg) {
 		case WM_SYSKEYUP:
 			break;
 		default:
-			return;
+			return false;
 	}
 
 	DWORD scancode = (msg->lParam >> 16) & 0xff;
 	DWORD vkcode = msg->wParam;
 	bool extended = !!(msg->lParam & 0x01000000);
 	bool up = !!(msg->lParam & 0x80000000);
 
-	qApp->postEvent(this, new InjectKeyboardMessageEvent(scancode, vkcode, extended, !up));
+	InjectMessageSuppressor suppressor;
+	InjectKeyboardMessageEvent *ikme = new InjectKeyboardMessageEvent(&suppressor, scancode, vkcode, extended, !up);
```

Comment:
If the handle functions were slots something like the following might also work:
```cpp
bool suppressed = false;
QMetaObject::invokeMethod(this, "handleKeyboardMessage", Qt::BlockingQueuedConnection, Q_RETURN_ARG(bool, suppressed), Q_ARG(DWORD, scancode), Q_ARG(DWORD, vkode), Q_ARG(bool, extended), Q_ARG(bool, up));
return suppressed;
```
With that we wouldn't have to mess with events at all and would get the syncronysed result for free. Then again the typing is horribly messed up and it is probably not as efficient ;) Ugly? Yes. Much less LOC? Yes. Good idea? Uhhh.....